### PR TITLE
Feat(messaging-api): use external sdk in backend

### PR DESCRIPTION
**Pull request by SamSalvatico**
### Description

Install external SDK in messaging-api

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### CODE REVIEW

File: apps/messaging-api/src/services/messages/messaging.ts [hunk 1]
```diff
@@ -382,7 +382,11 @@ export function newMessagingService(
       const schedulerSdk = await getSchedulerSdk(organisationId);
       const { error } = await schedulerSdk.scheduleTasks(scheduleBody);
       if (error) {
-        throw httpErrors.createError(503, error.message, { parent: error });
+        const errorDetail =
+          (error as { detail?: string }).detail ?? "Unknown error";
```
Comment (Thurfjell) [inline apps/messaging-api/src/services/messages/messaging.ts:386]:
```suggestion
          (error as { detail?: string })?.detail ?? "Unknown error";
```

File: apps/messaging-api/src/services/users/shared-users.ts [hunk 2]
```diff
@@ -305,12 +315,15 @@ export function ProfileSdkFacade(
...
-      const fromMessage = idsNotFound.length
+      const selectUsersResponse = idsNotFound.length
         ? await messagingProfile.selectUsers(idsNotFound)
         : [];
 
-      return { data: fromProfile.concat(fromMessage), error: undefined };
+      return {
+        data: fromProfile.concat(selectUsersResponse),
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        error: undefined as any,
```
Comment (Thurfjell) [inline apps/messaging-api/src/services/users/shared-users.ts:325]: Haha brutal 😂

File: apps/messaging-api/src/utils/authentication-factory.ts [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
-import { Profile, Scheduler, Upload } from "building-blocks-sdk";
+import getBuildingBlockSDK, {
+  BuildingBlocksSDK,
+  getM2MTokenFn,
+} from "@ogcio/building-blocks-sdk";
 import { httpErrors } from "@fastify/sensible";
 
+const trimSlash = (input: string) => {
+  let i = input.length;
+  while (i-- && input.charAt(i) === "/");
```
Comment (Thurfjell) [inline apps/messaging-api/src/utils/authentication-factory.ts:9]: Haha sneaky one liner. Does `i--` set i or do you need to do `--I`.

Comment (SamSalvatico) [reply apps/messaging-api/src/utils/authentication-factory.ts:9]: `i--` set minus after the value is used

File: apps/messaging-api/src/services/users/shared-users.ts (hunk 2 above)
Comment (SamSalvatico) [reply apps/messaging-api/src/services/users/shared-users.ts:325]: Unfortunately the types management of the open-api-definition library is not excellent/usable is some cases

File: apps/messaging-api/src/utils/authentication-factory.ts (hunk 3 above)
Comment (Thurfjell) [reply apps/messaging-api/src/utils/authentication-factory.ts:9]: I think this is confusing. Maybe skill issues. Go ahead I guess.

**Review by Thurfjell — APPROVED** [review]

**Pull request merged**